tectonic/core-bridge.c

```diff
@@ -110,6 +110,47 @@ bibtex_simple_main(tt_bridge_api_t *api, char *aux_file_name)
 
 #define TGB tectonic_global_bridge
 
+diagnostic_t
+ttstub_warn_begin(void)
+{
+    return TGB->warn_begin();
+}
+
+diagnostic_t
+ttstub_error_begin(void)
+{
+    return TGB->error_begin();
+}
+
+void
+ttstub_diag_finish(diagnostic_t diag)
+{
+    TGB->diag_finish(TGB->context, diag);
+}
+
+void
+ttstub_diag_append(diagnostic_t diag, char const *text)
+{
+    TGB->diag_append(diag, text);
+}
+
+void
+ttstub_diag_vprintf(diagnostic_t diag, const char *format, va_list ap)
+{
+    vsnprintf(error_buf, BUF_SIZE, format, ap); /* Not ideal to (ab)use error_buf here */
+    TGB->diag_append(diag, error_buf);
+}
+
+void
+ttstub_diag_printf(diagnostic_t diag, const char *format, ...)
```

Comment:
Can you please annotate this function with the `PRINTF_FUNC` macro? I see that you've used it in the header; TBH I'm not sure if that suffices or whether it also needs to go here, but it is already present for `ttstub_issue_warning` and friends.

Reply:
Whoops, looks like I just forgot. Thanks for catching this.